tensorflow/lite/micro/integration_tests/generate_per_layer_tests.py

```diff
@@ -0,0 +1,421 @@
+
+  def get_opcode_idx(self, builtin_operator):
+    for idx, opcode in enumerate(self.model.operatorCodes):
+      if schema_util.get_builtin_code_from_operator_code(opcode) == builtin_operator:
+        return idx
+
+  def generate_models(self, subgraph_idx, builtin_operator):
+    subgraph = self.model.subgraphs[subgraph_idx]
+    opcode_idx = self.get_opcode_idx(builtin_operator)
+    print(f'opcode_idx: {opcode_idx}')
+    output_models = []
+    for op in subgraph.operators:
+      if op.opcodeIndex == opcode_idx:
+        output_models.append(
+            self.generate_single_layer_model(self.model, subgraph, op,
+                                             opcode_idx))
+    return output_models
+
+
+class TestDataGenerator:
+
+  def __init__(self, output_dir, model_paths, inputs):
+    self.output_dir = output_dir
+    self.model_paths = model_paths
+    self.csv_filenames = []
+    self.inputs = inputs
+    self.cc_srcs = []
+    self.cc_hdrs = []
+    self.includes = []
+
+  def generate_inputs_conv(self, interpreter):
+    input_tensor = interpreter.tensor(
+        interpreter.get_input_details()[0]['index'])
+    return [
+        np.random.randint(
+            low=-128, high=127, dtype=np.int8, size=input_tensor().shape)
+    ]
+
+  def generate_goldens(self, builtin_operator):
+    for model_path in self.model_paths:
+      print(model_path)
+      # Load model and run a single inference with random inputs.
+      interpreter = tf.lite.Interpreter(model_path=model_path)
+      interpreter.allocate_tensors()
+      output_tensor = interpreter.tensor(
+          interpreter.get_output_details()[0]['index'])
+
+      if builtin_operator == schema_fb.BuiltinOperator.CONV_2D:
+        generated_inputs = self.generate_inputs_conv(interpreter)
+      else:
+        raise RuntimeError(f'Unsupported BuiltinOperator: {builtin_operator}')
+
+      print(generated_inputs[0])
+      for idx, input_tensor in enumerate(self.inputs):
+        interpreter.set_tensor(input_tensor, generated_inputs[idx])
+      interpreter.invoke()
+
+      for input_idx, input_tensor in enumerate(generated_inputs):
+        input_flat = input_tensor.flatten().tolist()
+        csv_input_filename = \
+            f"{model_path.split('.')[0]}_input{input_idx}_int8.csv"
+        input_csvfile = open(csv_input_filename, 'w', newline='')
+        input_csvwriter = csv.writer(input_csvfile)
+        input_csvwriter.writerow(input_flat)
+        self.csv_filenames.append(csv_input_filename)
+
+      output_flat = output_tensor().flatten().tolist()
+
+      # Write inputs and goldens to CSV file.
+      csv_golden_filename = f"{model_path.split('.')[0]}_golden_int8.csv"
+      golden_csvfile = open(csv_golden_filename, 'w', newline='')
+      golden_csvwriter = csv.writer(golden_csvfile)
+      np.set_printoptions(threshold=np.inf)
+      golden_csvwriter.writerow(output_flat)
+      self.csv_filenames.append(csv_golden_filename)
+
+  def generate_build_file(self):
```

Comment:
don't have to change it in this file, but I feel this file can benefit a lot from using mako. Basically, then all these boiler plate template string will be in a separate file and code will be much cleaner.   We can discuss this separately